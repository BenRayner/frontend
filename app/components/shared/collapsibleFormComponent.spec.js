/* global describe, it, jest, expect */
import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';

import collapsible from './collapsibleFormComponent';

describe('collapsibleFormComponent', () => {
  it('renders DOM element children as expected', () => {
    const CollapsibleFormComponent = collapsible('input');

    const component = TestUtils.renderIntoDocument(
      <CollapsibleFormComponent label="Test Input Field" />
    );

    const node = ReactDOM.findDOMNode(component); // eslint-disable-line react/no-find-dom-node
    expect(node).toMatchSnapshot();
  });

  it('renders React component children as expected', () => {
    const CollapsibleFormComponent = collapsible((props) => <pre>{JSON.stringify(props)}</pre>);

    const component = TestUtils.renderIntoDocument(
      <CollapsibleFormComponent label="Test Input Field" this-should-be="passed through" />
    );

    const node = ReactDOM.findDOMNode(component); // eslint-disable-line react/no-find-dom-node
    expect(node).toMatchSnapshot();
  });

  it('exposes a `focus` function', () => {
    const CollapsibleFormComponent = collapsible('input');

    const component = TestUtils.renderIntoDocument(
      <CollapsibleFormComponent label="Test Input Field" />
    );

    expect(Object.getOwnPropertyDescriptor(CollapsibleFormComponent.prototype, 'focus')).toMatchSnapshot();
    expect(() => component.focus()).not.toThrow();
  });

  it('exposes a `value` getter', () => {
    const CollapsibleFormComponent = collapsible('input');

    const component = TestUtils.renderIntoDocument(
      <CollapsibleFormComponent label="Test Input Field" defaultValue="meow" />
    );

    expect(Object.getOwnPropertyDescriptor(CollapsibleFormComponent.prototype, 'value')).toMatchSnapshot();
    expect(component.value).toBe("meow")
  });

  it('exposes specified proxy methods', () => {
    const CollapsibleFormComponent = collapsible('input');

    CollapsibleFormComponent.proxyMethods = ['select'];

    const component = TestUtils.renderIntoDocument(
      <CollapsibleFormComponent label="Test Input Field" />
    );

    expect(component.select).toBeInstanceOf(Function);
    expect(Object.getOwnPropertyDescriptor(component, 'select')).toMatchSnapshot();
    expect(() => component.select()).not.toThrow();
  });
});
