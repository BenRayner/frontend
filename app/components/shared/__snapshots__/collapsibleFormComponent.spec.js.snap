// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collapsibleFormComponent exposes a \`focus\` function 1`] = `
Object {
  "configurable": true,
  "enumerable": false,
  "value": [Function],
  "writable": true,
}
`;

exports[`collapsibleFormComponent exposes a \`value\` getter 1`] = `
Object {
  "configurable": true,
  "enumerable": false,
  "get": [Function],
  "set": undefined,
}
`;

exports[`collapsibleFormComponent exposes specified proxy methods 1`] = `
Object {
  "configurable": true,
  "enumerable": true,
  "value": [Function],
  "writable": true,
}
`;

exports[`collapsibleFormComponent renders DOM element children as expected 1`] = `
<div
  data-reactroot=""
  class="CollapsableFormComponent form-group CollapsableFormComponent--collapsed"
>
  <button
    type="button"
    class="CollapseExpandLinkComponent twbs-btn twbs-btn-link collapsed"
    data-toggle="collapse"
    data-target="#collapsable-form-1"
    aria-expanded="false"
    aria-controls="collapsable-form-1"
    style="padding-left: 0px;"
  >
    <strong
      class="semi-bold"
    >
      Test Input Field
    </strong>
    <i
      class="fa fa-angle-down"
      style="margin-left: 6px;"
    />
  </button>
  <div
    class="CollapsableFormComponent__content collapse"
    id="collapsable-form-1"
    aria-expanded="false"
  >
    <input
      class="input"
    />
    <!-- react-empty: 7 -->
    <!-- react-empty: 8 -->
  </div>
</div>
`;

exports[`collapsibleFormComponent renders React component children as expected 1`] = `
<div
  data-reactroot=""
  class="CollapsableFormComponent form-group CollapsableFormComponent--collapsed"
>
  <button
    type="button"
    class="CollapseExpandLinkComponent twbs-btn twbs-btn-link collapsed"
    data-toggle="collapse"
    data-target="#collapsable-form-2"
    aria-expanded="false"
    aria-controls="collapsable-form-2"
    style="padding-left: 0px;"
  >
    <strong
      class="semi-bold"
    >
      Test Input Field
    </strong>
    <i
      class="fa fa-angle-down"
      style="margin-left: 6px;"
    />
  </button>
  <div
    class="CollapsableFormComponent__content collapse"
    id="collapsable-form-2"
    aria-expanded="false"
  >
    <pre>
      {"this-should-be":"passed through","className":"input"}
    </pre>
    <!-- react-empty: 7 -->
    <!-- react-empty: 8 -->
  </div>
</div>
`;
